lib/kubernetes-deploy/task_config_validator.rb

```diff
@@ -70,10 +70,12 @@ def validate_namespace_exists
         return @errors << "Namespace can not be blank"
       end
 
-      _, err, st = @kubectl.run("get", "namespace", "-o", "name", namespace,
-        use_namespace: false, log_failure: false)
+      definition, err, st = @kubectl.run("get", "namespace", namespace,
+        use_namespace: false, log_failure: false, attempts: 3, output: 'json')
 
-      unless st.success?
+      if st.success?
+        @task_config.namespace_definition = JSON.parse(definition, symbolize_names: true)
```

Comment:
It isn't intuitive that `validate_namespace_exists` would have the side-effect of setting a property on an entirely different object. Could we make `task_config` capable of caching/returning the namespace object, and raising errors we can catch here?